tests/console/rmt/rmt_proxy.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: yast2-rmt rmt-server
+# Summary: Add rmt configuration test and basic configuration via
+#    rmt-wizard, test RMT basic function behind proxy.
+# Preparation: Setup the proxy
+# Select tinyproxy as example, on the machine to setup proxy server:
+#    steps: 1. zypper in tinyproxy
+#           2. update /etc/tinyproxy.conf:
+#              Set 'Port 8000'(can set your own port)
+#              Comments the 'Allow' to accept access
+#           3. systemctl restart tinyproxy
+# Then on RMT test machine, set /etc/rmt.conf as:
+#           proxy: http://{proxy server ip}:port
+#           proxy_auth: ntlm
+# Maintainer: Lemon Li <leli@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use repo_tools;
+use lockapi;
+use mmapi;
+use utils;
+
+sub run {
+    my $server_ip = "10.0.2.101";
+
+    select_console 'root-console';
+    record_info('RMT server setup', 'Start to setup a rmt server');
+    rmt_wizard();
+    # sync from SCC
+    rmt_sync;
+
+    # wait proxy server setup ready
+    mutex_wait("PROXY_SERVER_READY");
+
+    # Set RMT behind a proxy (with NTLM Auth), and ensure RMT basic function works.
+    record_info('Set proxy', 'Set RMT behind a proxy (with NTLM Auth)');
+    # set the proxy setting to rmt.conf
+    my $proxy_url = 'http:\/\/' . $server_ip . ':8000';
+    my $rmt_conf = '/etc/rmt.conf';
+    my $rmt_conf_bk = '/etc/rmt.conf.bk';
+
+    # backup the rmt conf file
+    assert_script_run("cp $rmt_conf $rmt_conf_bk");
+
+    # update the rmt conf file with the proxy setting
+    assert_script_run("sed -i 's/proxy: /proxy: $proxy_url/g' $rmt_conf");
+    assert_script_run("sed -i 's/proxy_auth: /proxy_auth: ntlm/g' $rmt_conf");
+    assert_script_run("systemctl restart rmt-server");
+
+    # verify the basic function of the rmt server
+    rmt_sync;
```

Comment:
Base https://openqa.nue.suse.com/tests/7849886#step/rmt_proxy/40, i suppose rmt-cli sync command already can verify the connection?
Use assert_script instead of rmt_sync to execute the command.

Reply:
Thanks, I updated code to use assert_script_run instead.

Verification run:  https://openqa.nue.suse.com/tests/7850193#step/rmt_proxy/40  proxy client
                         https://openqa.nue.suse.com/tests/7850192#details